vscode-dotnet-runtime-library/src/Acquisition/AcquisitionInvoker.ts

```diff
@@ -56,60 +61,76 @@ You will need to restart VS Code after these changes. If PowerShell is still not
+                let powershellReference = 'powershell.exe';
+                if (winOS)
                 {
-                    const powershellReference = await this.verifyPowershellCanRun(installContext, install);
+                    powershellReference = await this.verifyPowershellCanRun(installContext, install);
                     windowsFullCommand = windowsFullCommand.replace('powershell.exe', powershellReference);
                 }
 
                 cp.exec(winOS ? windowsFullCommand : installCommand,
-                        { cwd: process.cwd(), maxBuffer: 500 * 1024, timeout: 1000 * installContext.timeoutSeconds, killSignal: 'SIGKILL' },
-                        async (error, stdout, stderr) =>
-                {
-                    if (stdout)
+                    { cwd: process.cwd(), maxBuffer: 500 * 1024, timeout: 1000 * installContext.timeoutSeconds, killSignal: 'SIGKILL' },
+                    async (error, stdout, stderr) =>
                     {
+                        if (stdout)
+                        {
                             this.eventStream.post(new DotnetAcquisitionScriptOutput(install, TelemetryUtilities.HashAllPaths(stdout)));
-                    }
-                    if (stderr)
-                    {
+                        }
+                        if (stderr)
+                        {
                             this.eventStream.post(new DotnetAcquisitionScriptOutput(install, `STDERR: ${TelemetryUtilities.HashAllPaths(stderr)}`));
-                    }
-                    if (error)
-                    {
-                        if (!(await WebRequestWorker.isOnline(installContext.timeoutSeconds, this.eventStream)))
+                        }
+                        if (this.looksLikeBadExecutionPolicyError(stderr))
+                        {
+                            const badPolicyError = new EventBasedError('PowershellBadExecutionPolicy', `Your powershell execution policy does not allow script execution, so we can't automate the installation.
+Please read more at https:/go.microsoft.com/fwlink/?LinkID=135170`);
+                            this.eventStream.post(new PowershellBadExecutionPolicy(badPolicyError, install));
+                            reject(badPolicyError);
+                        }
+                        if ((this.looksLikeBadLanguageModeError(stderr) || error?.code === 1) && this.badLanguageModeSet(powershellReference))
+                        {
+                            const badModeError = new EventBasedError('PowershellBadLanguageMode', `Your Language Mode disables PowerShell language features needed to install .NET. Read more at: https://learn.microsoft.com/powershell/module/microsoft.powershell.core/about/about_language_modes?view=powershell-7.3.

```

Comment:
Url doesn't need specific PowerShell version in it.
```suggestion
                            const badModeError = new EventBasedError('PowershellBadLanguageMode', `Your Language Mode disables PowerShell language features needed to install .NET. Read more at: https://learn.microsoft.com/powershell/module/microsoft.powershell.core/about/about_language_modes.
```